### ISSUE TITLE
How is asserting None handled

### ISSUE THREAD

**Issue by humbienri** [Julian/jsonschema#603]
Given this requests response.json() (note the `None` instead of `null`):
```{
    "version": "2.3.4",
    "result": "Multi-Status",
    "data": [
        {
            "derId": 583,
            "warnings": [],
            "status": "success",
            "error": None
        }
    ]
}
```
 why is it that this schema isn't sufficient:
```{
    "$schema": "http://json-schema.org/draft-04/schema#",
    "title": "",
    "description": "JSON schema for the HTTP REST API response.",
    "type": "object",
    "properties": {
        "version": {"type": "string"},
        "result": {"type": "string"},
        "data": {
            "type": "array",
            "items": {
                "type": "object",
                "properties": {
                    "derId": {"type": "integer"},
                    "warnings": {"type": "array"},
                    "status": {"type": "string"},
                    "error": {"type": ["null","string"]}
                },
                "required": ["derId", "status", "warnings", "error"]
            }
        }
    },
    "required": ["data", "result", "version"]
}
```

Even if I remove the requirement for the error property, `jsonschema.validate()` chokes on the presence of `None`.

I realize that `None` is a python thing and not valid json, as I understand it, but it is the calling of the `.json()` api on the API's response that introduces the `None`.

Thank you in advance.

**Comment by Julian** [comment]
Hi, please include a full reproducer (the code you're running + its traceback).

Note though that, presuming you're talking about `requests`, that `requests.Response.json` does *not* return JSON, it returns Python objects, and that `jsonschema.validate` is expecting exactly that.

**Comment by humbienri** [comment]
Hello,
Sorry Julian, I did not include my actual code b/c it would get a bit tricky since my code is spread out across multiple modules and classes.  I guess I could doctor up an actual example to trigger this rather than modifying my code for illustrative purposes.

But I do think you've hit the nail on the head. Yes I am using python's `requests` but seems like you are saying that `*.json()` does not output valid json?

One thing I did notice is that no matter how much I fiddled with my code I kept seeing an error regarding `Assert None` and instructed to use `Assert is None`.  Your questions prompted me to try and reduce this down to just the commandline and try to reproduce this from there. So, `import jsonschema` and such and I was not able to reproduce the issue I was encountering.  Can't believe I spend days on this and it was totally my fault.

I think somewhere along the way I forgot that the `validate()` returns `None` on a successful validation?  So my `assert valid_per_schema(blah dih dah)` was essentially boiling down to `assert None' and it was pytest that was whining at me!!

Classic case of "we see what we want to see". Sheesh!

Thanks again @Julian! Sorry for the hassle.
